src/index.js

```diff
@@ -201,6 +201,61 @@ function EEAClient(web3, chainId) {
     });
   };
 
+  const distributeRawTransaction = options => {
+    if (options.privacyGroupId && options.privateFor) {
+      throw Error("privacyGroupId and privateFor are mutually exclusive");
+    }
+    const tx = new PrivateTransaction();
+    const privateKeyBuffer = Buffer.from(options.privateKey, "hex");
+    const from = `0x${privateToAddress(privateKeyBuffer).toString("hex")}`;
+    return web3.priv
+      .getTransactionCount({
+        from,
+        privateFrom: options.privateFrom,
+        privateFor: options.privateFor,
+        privacyGroupId: options.privacyGroupId
+      })
+      .then(transactionCount => {
+        tx.nonce = options.nonce || transactionCount;
+        tx.gasPrice = GAS_PRICE;
+        tx.gasLimit = GAS_LIMIT;
+        tx.to = options.to;
+        tx.value = 0;
+        tx.data = options.data;
+        // eslint-disable-next-line no-underscore-dangle
+        tx._chainId = chainId;
+        tx.privateFrom = options.privateFrom;
+
+        if (options.privateFor) {
+          tx.privateFor = options.privateFor;
+        }
+        if (options.privacyGroupId) {
+          tx.privacyGroupId = options.privacyGroupId;
+        }
+        tx.restriction = "restricted";
+        tx.sign(privateKeyBuffer);
+
+        const signedRlpEncoded = tx.serialize().toString("hex");
```

Comment:
This logic seems to be exactly the same as the one in the `sendRawTransaction` method. Can't we extract it to an auxiliary method or something?

The logic to build and sign a private tx can be reused, all that changes is the method that we call (either `eea_sendRawTransaction` or `priv_distrivuteRawTransaction`).